mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotation.java

```diff
@@ -84,6 +84,14 @@ private static boolean isValidKey(String input) {
         return true;
     }
 
+    public int countKeySlash() {
+        return countSlash(key);
+    }
+
+    public static int countSlash(String key) {
```

Comment:
Here I have several troubbles with this static method : 
 - low typing : you expect a Metadata key as an argument, but I can pass pretty anything : mail addresses, username, passwords... We kind of prefer to wrap the string in some other class to explicitly give it a name.
 - Responsability. Here you are exposing internals method that anybody can use, on anything. From the outside, is it my responsability to call this static method ?

What you truelly need is to define a class "MailboxAnnotationKey" : 
 - You can move there the key validation logic
 - You can count the number of Slash in it

And you don't need MailboxAnnotation::countKeySlash method anymore.
And every where you used to call MailboxAnnotation::countSlash you can first map to a MailboxAnnotationKey and then countSlash.

I guess you will end up with a cleaner schema for responsability, and definitly with stronger typing.